__tests__/integration/template/transaction/template.test.ts

```diff
@@ -0,0 +1,346 @@
+    expect(tx.outputs[7].tokenData).toBe(129);
+    expect(tx.outputs[7].value).toBe(2n);
+    expect(tx.outputs[11].tokenData).toBe(129);
+    expect(tx.outputs[11].value).toBe(2n);
+  });
+
+  it('should be able to send tokens and authorities', async () => {
+    const address = await hWallet.getAddressAtIndex(10);
+    const template = new TransactionTemplateBuilder()
+      .addSetVarAction({ name: 'addr', value: address })
+      .addSetVarAction({ name: 'token', value: tokenUid })
+      .addUtxoSelect({ fill: 2 })
+      .addTokenOutput({ address: '{addr}', amount: 2 })
+      .addUtxoSelect({ fill: 3, token: '{token}' })
+      .addTokenOutput({ address: '{addr}', amount: 3, token: '{token}' })
+      .addAuthoritySelect({ authority: 'mint', token: '{token}', count: 1 })
+      .addAuthorityOutput({ address: '{addr}', authority: 'mint', count: 1, token: '{token}' })
+      .addAuthoritySelect({ authority: 'melt', token: '{token}', count: 2 })
+      .addAuthorityOutput({ address: '{addr}', authority: 'melt', count: 2, token: '{token}' })
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+
+    expect(tx.outputs).toHaveLength(7);
+
+    // HTR Change
+    expect(tx.outputs[0].tokenData).toBe(0);
+    expect(tx.outputs[0].value).toBe(7n);
+
+    // HTR
+    expect(tx.outputs[1].tokenData).toBe(0);
+    expect(tx.outputs[1].value).toBe(2n);
+
+    // Custom token change
+    expect(tx.outputs[2].tokenData).toBe(1);
+    expect(tx.outputs[2].value).toBe(97n);
+
+    // Custom token
+    expect(tx.outputs[3].tokenData).toBe(1);
+    expect(tx.outputs[3].value).toBe(3n);
+
+    // mint authority
+    expect(tx.outputs[4].tokenData).toBe(129);
+    expect(tx.outputs[4].value).toBe(1n);
+
+    // melt authorities
+    expect(tx.outputs[5].tokenData).toBe(129);
+    expect(tx.outputs[5].value).toBe(2n);
+    expect(tx.outputs[6].tokenData).toBe(129);
+    expect(tx.outputs[6].value).toBe(2n);
+  });
+
+  it('should be able to destroy authorities', async () => {
+    const template = new TransactionTemplateBuilder()
+      .addSetVarAction({ name: 'token', value: tokenUid })
+      .addAuthoritySelect({ authority: 'mint', token: '{token}', count: 4 })
+      .addAuthoritySelect({ authority: 'melt', token: '{token}', count: 4 })
+      .build();
+
+    const tx = await interpreter.build(template, DEBUG);
+    await transactionUtils.signTransaction(tx, hWallet.storage, DEFAULT_PIN_CODE);
+    tx.prepareToSend();
+    const sendTx = new SendTransaction({ storage: hWallet.storage, transaction: tx });
+    await sendTx.runFromMining();
+    await waitForTxReceived(hWallet, tx.hash, null);
+
+    expect(tx.outputs).toHaveLength(0);
```

Comment:
suggestion: validate that the 8 inputs are there with the authorities

Reply:
Done